nc-news/src/components/RatingButtonManager.jsx

```diff
@@ -0,0 +1,121 @@
+import { useState, useEffect } from "react";
+import {
+  ActiveLikeButton,
+  ActiveDislikeButton,
+} from "./styled-components/StyledComponents";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faThumbsUp, faThumbsDown } from "@fortawesome/free-solid-svg-icons";
+
+const likesFromLocalStorage = JSON.parse(
+  localStorage.getItem("likedArticles") || "[]"
+);
+const dislikesFromLocalStorage = JSON.parse(
+  localStorage.getItem("dislikedArticles") || "[]"
+);
+
+export default function RatingButtonManager({ articleId, setVote }) {
+  const [likedArticles, setLikedArticles] = useState(likesFromLocalStorage);
+  const [dislikedArticles, setDislikedArticles] = useState(
+    dislikesFromLocalStorage
+  );
+  const [isLikeButtonActive, setIsLikeButtonActive] = useState(
+    likedArticles.length === 0 ? true : false
+  );
+  const [isDislikeButtonActive, setIsDislikeButtonActive] = useState(
+    dislikedArticles.length === 0 ? true : false
+  );
+
+  // Adding likes to local storage//
+  useEffect(() => {
+    localStorage.setItem("likedArticles", JSON.stringify(likedArticles));
+  }, [isLikeButtonActive]);
+
+  useEffect(() => {
+    localStorage.setItem("dislikedArticles", JSON.stringify(dislikedArticles));
+  }, [isDislikeButtonActive]);
+
+  // HandleClick Event //
+
+  const handleDislikeClick = (inc_vote) => {
+    setVote(inc_vote);
+    if (inc_vote === -1 && !dislikedArticles.includes(articleId)) {
+      setDislikedArticles((currDislikes) => {
+        return [...currDislikes, articleId];
+      });
+      setLikedArticles((currLikes) => {
+        return currLikes.filter((id) => id !== articleId);
+      });
+      setIsDislikeButtonActive(false);
+      setIsLikeButtonActive(true);
+    }
+    if (inc_vote === 1) {
+      setDislikedArticles((currDislikes) => {
+        return currDislikes.filter((id) => id !== articleId);
+      });
+      setIsDislikeButtonActive(true);
+    }
+  };
+
+  const handleLikeClick = (inc_vote) => {
+    setVote(inc_vote);
+    if (inc_vote === 1 && !likedArticles.includes(articleId)) {
+      setLikedArticles((currLikes) => {
+        return [...currLikes, articleId];
+      });
+      setDislikedArticles((currDislikes) => {
+        return currDislikes.filter((id) => id !== articleId);
+      });
+      setIsLikeButtonActive(false);
+      setIsDislikeButtonActive(true);
+    }
+    if (inc_vote === -1) {
+      setLikedArticles((currLikes) => {
+        return currLikes.filter((id) => id !== articleId);
+      });
+      setIsLikeButtonActive(true);
+    }
+  };
+
```

Comment:
Voting on articles is a good use case for [optimistic rendering ](https://notes.northcoders.com/courses/js-front-end/ui-behaviours-and-optimistic-rendering#optimistic-rendering)
One thing you need to do when you use optimistic rendering though is correct the state if the API request fails, which isn't currently happening. I.e. if the internet fails and the upvote doesn't go through, the error should be caught and result in correcting the state - more detail in the notes!